File: Aleshko_lab5/Aleshko_lab5/Aleshko_bank.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization.Formatters.Binary;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;

namespace Aleshko_lab5
{
    [Serializable]
    internal class Aleshko_bank
    {
        private List<Aleshko_account> bank = new List<Aleshko_account>();
        public void Addacc()
        {
            Aleshko_account acc;
            Console.WriteLine();
            acc = new Aleshko_account();
            acc.Input();
            bank.Add(acc);
        }
        public void AddVip()
        {
            Aleshko_vip acc;
            Console.WriteLine();
            acc = new Aleshko_vip();
            acc.Input();
            bank.Add(acc);
        }
        public void Print()
        {
            if (bank.Count == 0)
                Console.WriteLine("Банк банкрот ");
            else
            {
                Console.WriteLine("Клиенты банка: \n");
                foreach (Aleshko_account acc in bank)
                {
                    acc.Output();
                    Console.WriteLine("");
                }
            }
        }
        public void FileSave()
        {
            Console.WriteLine("Введите имя файла без расширения: ");
            string fileName = Console.ReadLine() + ".dat";
            using (FileStream fout = new FileStream(fileName, FileMode.OpenOrCreate))
            {
                BinaryFormatter formatter = new BinaryFormatter();
                formatter.Serialize(fout, bank); // перевод данных в байты
            }
        }
        public void FileLoad()
        {

            Console.WriteLine("Введите имя файла без расширения: ");
            string fileName = Console.ReadLine() + ".dat";
            using (FileStream fin = new FileStream(fileName, FileMode.Open))
            {
                BinaryFormatter formatter = new BinaryFormatter();
                bank = formatter.Deserialize(fin) as List<Aleshko_account>;
            }
        }
        public void Clear()
        {
            bank.Clear();
        }
    }
}

File: Aleshko_lab6/Aleshko_dll/Aleshko_bank.cpp
#include "pch.h"
#include "Aleshko_bank.h"
#include "Aleshko_account.h"
#include "Aleshko_vip.h"



void Aleshko_bank::push(std::shared_ptr<Aleshko_account> acc)
{
    accounts.push_back(acc);
}



void Aleshko_bank::clear() {
    accounts.clear();
}

void Aleshko_bank::WriteToFile(CArchive& ar) const
{
    ar << (int)accounts.size();
    for (const auto& acc : accounts) {
        ar << acc.get();
    }
}

void Aleshko_bank::ReadFromFile(CArchive& ar)
{
    int n;
    Aleshko_account* acc;
    ar >> n;
    for (int i = 0; i < n; ++i) {
        ar >> acc;
        push(std::shared_ptr<Aleshko_account>(acc));
    }
}
